### ISSUE TITLE
[Feature]: Support for Cloudflare Browser Rendering API

### ISSUE THREAD

**Issue by dpeek** [microsoft/playwright#32042]
### 🚀 Feature Request

Cloudflare offer a simple and very const effective "serverless" chromium in the form of the [browser rendering API](https://developers.cloudflare.com/browser-rendering).

I'd love to use the far superior Playwright API to control this rather than Puppeteer (which they [forked](https://github.com/cloudflare/puppeteer) to run in a non-node environment).

Would it be technically possible run the Playwright "client" to run in a non-node runtime? I naively took a look myself but it seemed that Playwright is very much tied to Node.

More than happy to put the work in, but was hoping for some feedback on whether this is a fools errand or not :)

### Motivation

This would allow us to run our Playwright automations on Cloudflare. As a side benefit you'd basically be able to run the client API in _any_ JS runtime?

**Comment by dpeek** [comment]
Yes, just the playwright library. Cloudflare does provide a compat layer for many Node APIs, but in my quick investigation it seemed very hard to separate the "server" part (which uses a bunch of Node APIs that they don't support) from the client part (which seems like it should be supportable).

What node dependencies would the client have right now? Is devtools over HTTP/WS? Perhaps an easier way forward would be for me to use one of the other runtime ports of the client as a reference – although it seems wasteful to not use the existing JS client impl as a foundation.

**Comment by yury-s** [comment]
> What node dependencies would the client have right now? 

It's 'fs', 'path' for filesystem access, zip and some network stuff (may be stripped off in many scenarios), you can look at the [client](https://github.com/microsoft/playwright/tree/main/packages/playwright-core/src/client) source and corresponding dependencies from `DEPS.list`.

> Is devtools over HTTP/WS? Perhaps an easier way forward would be for me to use one of the other runtime ports of the client as a reference – although it seems wasteful to not use the existing JS client impl as a foundation.

`connectOverCDP` is implemented on the [server](https://github.com/microsoft/playwright/tree/main/packages/playwright-core/src/server) side, so you'll need that one, you cannot just take the client without server. The server dependencies should be mostly limited to fs, process launch API (which you shouldn't need when connecting over CDP), zip and websocket/network. It shouldn't be hard to make it run without filesystem dependencies, if you don't need to load/save any artifacts.

FYI, there was an attempt to run Playwright as an extension: https://github.com/microsoft/playwright/pull/24020, you might find it helpful.

**Comment by ruifigueira** [comment]
Hi there, just to let you know that there's already a [playwright fork that runs in Cloudfrare Browser Rendering](https://github.com/cloudflare/playwright).
